src/host/messaging.ts

```diff
@@ -13,7 +13,14 @@ export function initializeMessaging() {
     });
 
     window.addEventListener("message", (messageEvent) => {
-        if (messageEvent.origin === "vscode-resource://") {
+        // Both windows now have a "null" origin so we need to distiguish direction based on protocol,
+        // which will throw an exception when it is from the devtools x-domain window.
+        let sendToDevTools = false;

```

Comment:
Wouldn't rely on this approach will be super expensive? (relying on exception handling as logic for the code)

Reply:
Unfortunately I can't think of another way around this. Since the origin is now "null" for both the data:// and vscode-resource:// origins, the only way to tell which it is, is to check something on the window (source) object. But that will throw an exception when it is pointing at the iframe. This is only true for VSCode > 140 which is why it actually compares the protocol so that in <140 it will not throw and still know the direction.

Reply:
can we create an issue and link that in here with a comment, I would like to come back in the future, were hopefully there is a fix for this and address the issue.

Reply:
Opened #110